File: 2116_check_if_a_parentheses_string_can_be_valid.py
class Solution(object):
    def canBeValid(self, a, b):
        c = len(a)
        if c % 2: return False
        d = 0
        for e in range(c):
            d += 1 if a[e] == '(' or b[e] == '0' else -1
            if d < 0: return False
        d = 0
        for e in range(c - 1, -1, -1):
            d += 1 if a[e] == ')' or b[e] == '0' else -1
            if d < 0: return False
        return True

File: 2116_check_if_a_parentheses_string_can_be_valid.cpp
class Solution {
public:
    bool canBeValid(string a, string b) {
        int c = a.size(), d = 0;
        if (c % 2) return 0;
        for (int e = 0; e < c; e++) {
            d += (a[e] == '(' || b[e] == '0') ? 1 : -1;
            if (d < 0) return 0;
        }
        d = 0;
        for (int e = c - 1; e >= 0; e--) {
            d += (a[e] == ')' || b[e] == '0') ? 1 : -1;
            if (d < 0) return 0;
        }
        return 1;
    }
};
